boa/util/abi.py

```diff
@@ -1,23 +1,30 @@
 # wrapper module around whatever encoder we are using
+from typing import Any
 
 from eth.codecs.abi.decoder import Decoder
 from eth.codecs.abi.encoder import Encoder
+from eth.codecs.abi.nodes import ABITypeNode
 from eth.codecs.abi.parser import Parser
+from eth_abi import is_encodable
 
-_parsers: dict[str, Parser] = {}
+_parsers: dict[str, ABITypeNode] = {}
 
 
-def _get_parser(schema):
+def _get_parser(schema: str):
     try:
         return _parsers[schema]
     except KeyError:
         _parsers[schema] = (ret := Parser.parse(schema))
         return ret
 
 
-def abi_encode(schema, data):
+def abi_encode(schema: str, data: Any) -> bytes:
     return Encoder.encode(_get_parser(schema), data)
 
 
-def abi_decode(schema, data):
+def abi_decode(schema: str, data: bytes) -> Any:
     return Decoder.decode(_get_parser(schema), data)
+
+
+def is_abi_encodable(abi_type: str, data: Any) -> bool:
+    return is_encodable(abi_type, data)
```

Comment:
how does `is_encodable` work?

Reply:
Well this function is just a wrapper around the eth_abi function as you can see. 
However I'm looking into their implementation and they just try and catch :sweat: 
So I think it makes sense to stick with eth.codecs and do the try/catch ourselves